src/column_family.rs

```diff
@@ -28,24 +29,64 @@ pub const DEFAULT_COLUMN_FAMILY_NAME: &str = "default";
 pub struct ColumnFamilyDescriptor {
     pub(crate) name: String,
     pub(crate) options: Options,
+    pub(crate) ttl: CfTtl,
 }
 
 impl ColumnFamilyDescriptor {
-    // Create a new column family descriptor with the specified name and options.
+    /// Create a new column family descriptor with the specified name and options.
+    /// *WARNING*:
+    /// Will use [`CfTtl::SameAsWholeDb`] as ttl.
     pub fn new<S>(name: S, options: Options) -> Self
     where
         S: Into<String>,
     {
         Self {
             name: name.into(),
             options,
+            ttl: CfTtl::SameAsWholeDb,
         }
     }
 
+    /// Create a new column family descriptor with the specified name, options, and ttl.
+    /// *WARNING*:
+    /// The ttl is applied only when DB is opened with [`DB::open_with_ttl()`].
+    pub fn new_with_ttl<S>(name: S, options: Options, ttl: CfTtl) -> Self
+    where
+        S: Into<String>,
+    {
+        Self {
+            name: name.into(),
+            options,
+            ttl,
+        }
+    }
+
+    /// Sets ttl for the column family. It's applied only when DB is opened with
+    /// [`DB::open_with_ttl()`]. Changing ttl after DB is opened has no effect.
+    pub fn set_ttl(&mut self, ttl: CfTtl) {
+        self.ttl = ttl;
+    }
+
     /// Get the name of the ColumnFamilyDescriptor.
     pub fn name(&self) -> &str {
         &self.name
     }
+
+    pub fn ttl(&self) -> CfTtl {
+        self.ttl
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// Specifies the TTL behavior for a column family.
+/// https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46
+pub enum CfTtl {
```

Comment:
I would suggest to rename to the `ColumnFamilyTtl` to be consistent. E.g. `ColumnFamilyDescriptor`

Reply:
Done